src/braket/circuits/angled_gate.py

```diff
@@ -114,3 +115,124 @@ def __eq__(self, other):
+                printing a diagram of a circuit. The length must be the same as `qubit_count`, and
+                index ordering is expected to correlate with the target ordering on the instruction.
+                For instance, if a CNOT instruction has the control qubit on the first index and
+                target qubit on the second index, the ASCII symbols should have `["C", "X"]` to
+                correlate a symbol with that index.
+
+        Raises:
+            ValueError: If the `qubit_count` is less than 1, `ascii_symbols` are `None`, or
+                `ascii_symbols` length != `qubit_count`, or `angle` is `None`
+        """
+        super().__init__(qubit_count=qubit_count, ascii_symbols=ascii_symbols)
+        if angle_1 is None or angle_2 is None:
+            raise ValueError("angles must not be None")
+        self._parameters = [
+            (
+                angle
+                if isinstance(angle, FreeParameterExpression)
+                else float(angle)  # explicit casting in case angle is e.g. np.float32
+            )
+            for angle in (angle_1, angle_2)
+        ]
+
+    @property
+    def parameters(self) -> List[Union[FreeParameterExpression, float]]:
+        """
+        Returns the parameters associated with the object, either unbound free parameters or
+        bound values.
+
+        Returns:
+            List[Union[FreeParameterExpression, float]]: The free parameters or fixed value
+            associated with the object.
+        """
+        return self._parameters
+
+    @property
+    def angle_1(self) -> Union[FreeParameterExpression, float]:
+        """
+        Returns the first angle for the gate
+
+        Returns:
+            Union[FreeParameterExpression, float]: The first angle of the gate in radians
+        """
+        return self._parameters[0]
+
+    @property
+    def angle_2(self) -> Union[FreeParameterExpression, float]:
+        """
+        Returns the second angle for the gate
+
+        Returns:
+            Union[FreeParameterExpression, float]: The second angle of the gate in radians
+        """
+        return self._parameters[1]
+
+    def bind_values(self, **kwargs) -> AngledGate:
+        """
+        Takes in parameters and attempts to assign them to values.
+
+        Args:
+            **kwargs: The parameters that are being assigned.
+
+        Returns:
+            AngledGate: A new Gate of the same type with the requested parameters bound.
+
+        Raises:
+            NotImplementedError: Subclasses should implement this function.
+        """
+        raise NotImplementedError
+
+    def adjoint(self) -> List[Gate]:
+        """Returns the adjoint of this gate as a singleton list.
+
+        Returns:
+            List[Gate]: A list containing the gate with negated angle.
+        """
+        raise NotImplementedError
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, DoubleAngledGate)
```

Comment:
Two double angle gates with the same parameters will not in general be the same
```suggestion
            isinstance(other, type(self))
```

Reply:
This isn't a criteria for equality as much as a short-circuit to ensure the actual comparisons don't error out. That said, I did forget to check name